antismash/common/pfamdb.py

```diff
@@ -113,6 +118,27 @@ def _build_mapping(database: str) -> Dict[str, str]:
     return mapping
 
 
+def get_database_type(database: str) -> str:
```

Comment:
I agree that this is a crude solution just to avoid a logging message, but I'm not really sure how to apply your suggestions. 
The problem arises because we're generating an HmmerResults object using a non-Pfam database, which is not done anywhere else within antiSMASH. This doesn't really matter for the most part, except for when it can't figure out what Pfam version it's working with (which it only needs for the logging). 

Maybe we could generalize `antismash.common.hmmer.py` a bit to also take in other databases? That would make it more friendly for future modules too.
We could then make the distinction between a pfam-database and another type of database early on in the `run_hmmer` function call based on database name.

It only really matters for that single logging line, though.